### PR TITLE
Adds GIFs

### PR DESCRIPTION
**Pull request by ksatirli**
Jiffs, Giphs - whatever you want to call it, I added an animated image of the provider mirrors feature and did some more clean-up.

### CODE REVIEW

File: terraform-providers-mirror/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+> Provider binaries may be mirrored to a local filesystem (or a networked HTTP server) and can be used as an alternative source of installation when directly accessing the origin registries is undesirable.
+
+* Terraform documentation: [Provider installation](https://www.terraform.io/docs/commands/cli-config.html#provider-installation)
+* GitHub PR / Issue: [#25999](https://github.com/hashicorp/terraform/issues/25999)
+
+---
+
+## Demonstration
+
+![terraform providers mirror](terraform-providers-mirror.gif)
```
Comment (onlydole) [inline terraform-providers-mirror/README.md:14]: This isn't working in preview yet, but should once this gets merged (noting for posterity) 😄

File: terraform-providers-mirror/.terraform.lock.hcl [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# This file is maintained automatically by "terraform init".
```
Comment (onlydole) [inline terraform-providers-mirror/.terraform.lock.hcl:1]: This lock file might not be necessary if this is used for TF 0.13.x examples

**Review by onlydole — CHANGES_REQUESTED** [review]
Some small nits to double-check, otherwise looks great to me 👍🏼

File: terraform-providers-mirror/.terraform.lock.hcl (hunk 2 above)
Comment (ksatirli) [reply terraform-providers-mirror/.terraform.lock.hcl:1]: This is a very valid point!

File: terraform-providers-mirror/README.md (hunk 1 above)
Comment (ksatirli) [reply terraform-providers-mirror/README.md:14]: Yeah it is a tad tricky, I find!

**Review by onlydole — APPROVED** [review]
Glorious 🙌🏼 

phenomenal PR

**Pull request merged**